src/command_modules/azure-cli-interactive/tests/test_feedback.py

```diff
@@ -0,0 +1,71 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import datetime
+import unittest
+
+import azclishell.frequency_heuristic as fh
+from azclishell.configuration import CONFIGURATION, get_config_dir as shell_config
+
+SHELL_CONFIG = CONFIGURATION
+
+
+def _mock_update():
+    return {fh.day_format(datetime.datetime.utcnow()): 1}
+
+
+def _mock_update2():
+    return {
+        fh.day_format(datetime.datetime.utcnow()): 2,
+        fh.day_format(datetime.datetime.utcnow() - datetime.timedelta(days=2)): 1}
+
+
+def _mock_update3():
+    return {
+        fh.day_format(datetime.datetime.utcnow()): 19,
+        fh.day_format(datetime.datetime.utcnow() - datetime.timedelta(days=18)): 5,
+        fh.day_format(datetime.datetime.utcnow() - datetime.timedelta(days=27)): 2,
+        fh.day_format(datetime.datetime.utcnow() - datetime.timedelta(days=28)): 2,
+        fh.day_format(datetime.datetime.utcnow() - datetime.timedelta(days=100)): 1,
+        fh.day_format(datetime.datetime.utcnow() - datetime.timedelta(days=200)): 1}
+
+
+class FeedbackTest(unittest.TestCase):
+    """ tests the frequncy heuristic """
+    def __init__(self, *args, **kwargs):
+        super(FeedbackTest, self).__init__(*args, **kwargs)
+        self.norm_update = fh.update_frequency
+
+    def test_heuristic(self):
+        """ test the correct logging of time for frequency """
+        fh.update_frequency = _mock_update
+        self.assertEqual(1, fh.frequency_measurement())
+
+        fh.update_frequency = _mock_update2
+        self.assertEqual(2, fh.frequency_measurement())
+
+        fh.update_frequency = _mock_update3
+        self.assertEqual(3, fh.frequency_measurement())
+
+    def test_update_freq(self):
+        """ tests updating the files for frequency """
+        fh.update_frequency = self.norm_update
+
+        freq_path = os.path.join(shell_config(), SHELL_CONFIG.get_frequency())
```

Comment:
You need to mock the file path.
The test can't be run on the real config, especially if you remove it.

This would mean if I run these tests on my box, the test would delete my real frequency file?